src/main/java/org/jenkinsci/plugins/workflow/libs/LibraryAdder.java

```diff
@@ -190,14 +190,15 @@
 
             if(versionCacheDir.exists()) {
                 listener.getLogger().println("Library " + name + "@" + version + " is cached. Copying from home.");
-                lastReadFile.touch(System.currentTimeMillis());
             } else {
                 listener.getLogger().println("Caching library " + name + "@" + version);
                 versionCacheDir.mkdirs();
                 retrieveLockFile.touch(System.currentTimeMillis());
                 retriever.retrieve(name, version, changelog, versionCacheDir, run, listener);
                 retrieveLockFile.delete();
             }
+            lastReadFile.touch(System.currentTimeMillis());
```

Comment:
Now we write this even when creating the cache directory for the first time, so that it will eventually be deleted even if it was only used once.